xcube/core/tile.py

```diff
@@ -261,37 +262,15 @@ def _get_var_2d_array(var: xr.DataArray,
         if labels_are_indices:
             array = var.isel(**labels)
         else:
-            array = var.sel(method='nearest', **labels)
+            safe_labels = ensure_time_label_compatible(var, labels)
+            array = var.sel(method='nearest', **safe_labels)
```

Comment:
`xcube/core/tile.py` will be deprecated. For xcube server NG, a lot of code changes have been introduced. After merging this PR, please update also https://github.com/dcs4cop/xcube/pull/677.